libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -461,8 +478,17 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 		}
 	}
 
-	m_assembly.appendLabel(m_context->functionExitPoints.top().label);
-	m_context->functionExitPoints.pop();
+	if (!subTransform.m_functionExitStackHeight.has_value())
```

Comment:
Maybe add a helper function (e.g. `returnVariablesAreSetup()`) to make it clear what this condition means?

Reply:
Or maybe not, the following code is all about stack height...

Reply:
> Or maybe not, the following code is all about stack height...

Only read that after I added the rather bulky ``returnVariablesAndFunctionExitAreSetup`` helper :-). But I guess it's fine that way.